src/Symfony/Component/Cache/DependencyInjection/CachePoolPass.php

```diff
@@ -116,35 +116,31 @@ public function process(ContainerBuilder $container)
 
             if (ChainAdapter::class === $class) {
                 $adapters = [];
-                foreach ($adapter->getArgument(0) as $provider => $adapter) {
-                    if ($adapter instanceof ChildDefinition) {
-                        $chainedPool = $adapter;
-                    } else {
-                        $chainedPool = $adapter = new ChildDefinition($adapter);
+                foreach ($adapter->getArgument(0) as $provider => $chainableAdapter) {
```

Comment:
no need to rename, especially because that's going to create needless merge conflicts in the future.

Reply:
That `process` method is really long and hard to grasp what it's doing. What if `$adapter` gets called outside this foreach in the future? Or is that too less likely to happen?

Reply:
we shall have tests to spot issues

Reply:
Ok, added daa6e4b to revert renaming the variable